### PR TITLE
js-oop frogger-game

### PR DESCRIPTION
**Pull request by iamsokol**
### js-oop

[Demo](https://iamsokol.github.io/frogger-arcade-game/)  
[Code base](https://github.com/iamsokol/iamsokol.github.io/tree/master/frogger-arcade-game)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by OleksiyRudenko** [comment]
@iamsokol 
This is not a code review but rather some impressions from demo.
When the hero stays on either of two topmost rows (marked with red rect on a screenshot below) and an enemy runs one row below the hero gets reset on enemy proximity.

image


The game also allows the hero to reach outside the playfield. Moreover the hero gets reset once outside the playfield. Not really clear if it is a win or not.

**Comment by OleksiyRudenko** [comment]
@iamsokol I took a look at your code line 3, which says `const ENEMIES_POSITION = [50, 130, 220],`. 
The rows are all of equal height actually, while 130 - 50 === 80 and 220 - 130 = 90. This might be a source of somewhat unexpected behaviour.

File: submissions/iamsokol/js-oop/app.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Player.prototype.restartPlayer = function () {
+    player.x = PLAYER_X;
+    player.y = PLAYER_Y
+};
+
+Player.prototype.handleInput = function(pressedKey) {
+    switch (pressedKey) {
+        case 'left':
+            this.x -= this.motion + 50;
```
Comment (OleksiyRudenko) [inline submissions/iamsokol/js-oop/app.js]: 50 (and 30 below, and also some other numbers across the code) is a mysterious number. Please, demystify defining it as a constant with a semantic name, just like you did for many other numbers.
We write code for other people, firsthand. Please, take care of possible future contributors to your code.

File: submissions/iamsokol/js-oop/app.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        this.x = -WIDTH_BLOCK;
+    }
+    this.checkCollisions();
+}
+
+Enemy.prototype.checkCollisions = function(dt) {
+    let position1 = player.x < this.x + WIDTH_PLAYER,
+        position2 = player.x + WIDTH_PLAYER > this.x,
+        position3 = player.y < this.y + HEIGHT_PLAYER,
+        position4 = player.y + HEIGHT_PLAYER > this.y;
```
Comment (OleksiyRudenko) [inline submissions/iamsokol/js-oop/app.js]: Classes should be agnostic of the outer world. The only globals allowed to refer to from within the class are global constants and library functions.
Other entities should be known to some class when such class is explicitly informed about such entity.
Please, fix.

**Comment by iamsokol** [comment]
@OleksiyRudenko, I fixed everything

**Pull request merged**